pinot-controller/src/test/java/org/apache/pinot/controller/api/PinotTableRestletResourceTest.java

```diff
@@ -273,6 +273,33 @@ public void rebalanceTableWithoutSegments()
         RebalanceResult.class);
     Assert.assertEquals(rebalanceResult.getStatus(), RebalanceResult.Status.NO_OP);
   }
+  
+  @Test
+  public void testTruncateTable() throws IOException {
```

Comment:
we need some more testing
1. create a OFFLINE table and schema, add some segments, then call truncate. Check that segments/idealstate/external view are deleted, but the tableconfig and schema are intact
2. create a REALTIME table and schema, let it create some CONSUMING segments, then call truncate. Check that table config and schema re intact. Also check that idealstate/external view/segments now have the newly created segments.